lib/spack/spack/test/llnl/util/filesystem.py

```diff
@@ -484,3 +484,82 @@ def test_filter_files_multiple(tmpdir):
         assert '<malloc.h>' not in f.read()
         assert '<string.h>' not in f.read()
         assert '<stdio.h>' not in f.read()
+
+
+# Each test input is a tuple of entries which prescribe
+# - the 'subdirs' to be created from tmpdir
```

Comment:
I would have expected this to just be specified like "x/y" but I'm guessing you are doing this because `tmpdir.ensure` wouldn't work if you passed it `x/y`. To make this clearer, could you have a single argument (i.e. use "x/y" as a parameter instead of ["x", "y"]) but then do something like

```
path.split(os.pathsep)
```

which would decompose the path into the individual directory args.

Reply:
> but I'm guessing you are doing this because tmpdir.ensure wouldn't work if you passed it x/y

That's correct. I'm writing tests parameters like that because they're ready to be consumed by `tmpdir.ensure`. I'll change like you said (though I'll split on `/` and not on `os.sep` so that I don't have to construct the parameter with `os.path.join`, which would defeat the purpose of added clarity).

Reply:
Done in 8982ee4